### PR TITLE
[ALS-4774] Add requester email to query request

### PR DESCRIPTION
**Pull request by Luke-Sikina**

### CODE REVIEW

File: pic-sure-api-war/src/main/java/edu/harvard/dbmi/avillach/service/PicsureQueryService.java [hunk 1]
```diff
@@ -264,141 +257,138 @@ public QueryStatus queryMetadata(UUID queryId, HttpHeaders headers){
...
-		Resource resource = verifyQueryRequest(dataQueryRequest, headers);
-		dataQueryRequest.getResourceCredentials().put(ResourceWebClient.BEARER_TOKEN_KEY, resource.getToken());
-
-		QueryStatus response = resourceWebClient.query(resource.getResourceRSPath(), dataQueryRequest);
-		Query queryEntity = copyQuery(dataQueryRequest, resource, response);
-		queryRepo.persist(queryEntity);
-		// we don't want the user to see the common area ID for now, but this could be useful later
-		// for editing the query
-		response.getResultMetadata().put("commonAreaId", queryEntity.getUuid().toString());
-
```
Comment (Luke-Sikina) [inline pic-sure-api-war/src/main/java/edu/harvard/dbmi/avillach/service/PicsureQueryService.java:298]: This is the only non formatting change in this file

File: pic-sure-api-war/src/main/java/edu/harvard/dbmi/avillach/service/PicsureQueryService.java [hunk 2]
```diff
@@ -264,141 +257,138 @@ public QueryStatus queryMetadata(UUID queryId, HttpHeaders headers){
...
+            response.setResultMetadata(new HashMap<>());
+        }
+        Map<String, Object> metaData = response.getResultMetadata();
+
+        if (dataQueryRequest instanceof FederatedQueryRequest) {
+            FederatedQueryRequest gicRequest = (FederatedQueryRequest) dataQueryRequest;
+            metaData.put("commonAreaUUID", gicRequest.getCommonAreaUUID());
+            metaData.put("site", gicRequest.getInstitutionOfOrigin());
+            metaData.put("sharingStatus", DataSharingStatus.Unknown);
+            metaData.put("requesterEmail", gicRequest.getRequesterEmail());
```
Comment (Luke-Sikina) [inline pic-sure-api-war/src/main/java/edu/harvard/dbmi/avillach/service/PicsureQueryService.java:317]: Non formatting change

File: pic-sure-api-war/src/main/java/edu/harvard/dbmi/avillach/service/PicsureQueryService.java [hunk 3]
```diff
@@ -264,141 +257,138 @@ public QueryStatus queryMetadata(UUID queryId, HttpHeaders headers){
...
+        if (dataQueryRequest.getQuery() != null) {
+            try {
+                queryJson = mapper.writeValueAsString(dataQueryRequest);
+            } catch (JsonProcessingException e) {
+                throw new ProtocolException(ProtocolException.INCORRECTLY_FORMATTED_REQUEST);
+            }
+        }
+        if (response.getResultMetadata() == null) {
+            response.setResultMetadata(new HashMap<>());
+        }
```
Comment (Luke-Sikina) [inline pic-sure-api-war/src/main/java/edu/harvard/dbmi/avillach/service/PicsureQueryService.java:309]: Non formatting change

File: pic-sure-resources/pic-sure-resource-api/src/main/java/edu/harvard/dbmi/avillach/domain/FederatedQueryRequest.java [hunk 4]
```diff
@@ -6,117 +6,112 @@
...
+    example = "{\n" + "    \"resourceUUID\": \"<RESOURCE UUID>\",\n" + "    \"query\": {\n" + "        \"categoryFilters\": {\n"
+        + "            \"\\\\demographics\\\\SEX\\\\\": [\n" + "                \"female\",\n" + "                \"male\"\n"
+        + "            ]\n" + "        },\n" + "        \"numericFilters\": {\n" + "            \"\\\\demographics\\\\AGE\\\\\": {\n"
+        + "                \"min\": \"0\",\n" + "                \"max\": \"85\"\n" + "            }\n" + "        },\n"
+        + "        \"requiredFields\": [],\n" + "        \"anyRecordOf\": [\n" + "            \"\\\\demographics\\\\RACE\\\\\"\n"
+        + "        ],\n" + "        \"variantInfoFilters\": [\n" + "            {\n"
+        + "                \"categoryVariantInfoFilters\": {},\n" + "                \"numericVariantInfoFilters\": {}\n"
+        + "            }\n" + "        ],\n" + "        \"expectedResultType\": \"DATAFRAME\",\n" + "        \"fields\": []\n" + "    }\n"
+        + "}"
+)
```
Comment (Gcolon021) [inline pic-sure-resources/pic-sure-resource-api/src/main/java/edu/harvard/dbmi/avillach/domain/FederatedQueryRequest.java:23]: I think the formatter made things worse here. Not sure if we can add some syntax that allows us to skip formatting a section.

**Review by Gcolon021 — APPROVED** [review]

**Pull request merged**